hack/setup_env.sh

```diff
@@ -15,9 +19,23 @@ function print_help() {
   echo "Usage: bash ${0} (${ALL_FUNCS})"
 }
 
+function download() {
+    # This function will leverage buildkit cache, and avoid downloading the file again
+    # if we already have a file with the same name in the cache directory
```

Comment:
can't we do the check in the target destination of the file?
I mean, we can consider the target path as a "cache" in a way, and remove the need of file duplications

Reply:
maybe it will lack the versioning impact in case someone changes the binary version, but seems good to go simpler at first than have a more complex logic

Reply:
ack